### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by floriangeyer21**

### CODE REVIEW

File: src/main/webapp/WEB-INF/views/accessDenied.jsp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    </style>
+</head>
+<body class="body1">
+<link href="${pageContext.request.contextPath}/resources/style4.css" rel="stylesheet" type="text/css">
+<h1 class="font">Sorry, you don't have permission access to this page.</h1>
+<div class="font : font1">
+    <a href="${pageContext.request.contextPath}/">To main page</a>
+</div>
+</body>
+</html>
```
Comment (DmGrt) [inline src/main/webapp/WEB-INF/views/accessDenied.jsp]: Add empty line)

**Review by DmGrt — APPROVED** [review]
Good job! just add one empty line

File: src/main/java/com/internet/shop/controllers/InjectDataController.java [hunk 2]
```diff
@@ -27,16 +29,25 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
...
+        User admin = new User("admin", "admin", "345684");
+        user1.setRoles(Set.of(Role.of("USER")));
+        user2.setRoles(Set.of(Role.of("USER")));
+        admin.setRoles(Set.of(Role.of("ADMIN")));
         userService.create(user1);
         userService.create(user2);
+        userService.create(admin);
+        ShoppingCart shoppingCart1 = new ShoppingCart(user1.getId());
+        ShoppingCart shoppingCart2 = new ShoppingCart(user2.getId());
+        ShoppingCart shoppingCart3 = new ShoppingCart(admin.getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/InjectDataController.java]: do we need shoppingCart for admin?

File: src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    private Map<String, List<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/users/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/add", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/manage", List.of(Role.RoleName.ADMIN));
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java]:
```suggestion
        protectedUrls.put("/product/manage", List.of(Role.RoleName.ADMIN));
```

File: src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        if (isAuthorized(user, protectedUrls.get(requestUrl))) {
+            filterChain.doFilter(req, resp);
+        } else {
+            req.getRequestDispatcher("/WEB-INF/views/accessDenied.jsp").forward(req, resp);
+        }
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java]:
```suggestion
```

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**